core/server/master/src/main/java/alluxio/underfs/UfsStatusCache.java

```diff
@@ -152,15 +153,15 @@ public UfsStatus fetchStatusIfAbsent(AlluxioURI path, MountTable mountTable)
       UnderFileSystem ufs = ufsResource.get();
       UfsStatus ufsStatus = ufs.getStatus(ufsUri.toString());
       if (ufsStatus == null) {
-        return null;
+        throw new FileNotFoundException("fetched status is null for: " + ufsUri);
```

Comment:
Will this exception be thrown to the client? If so we should make it more explanatory (ie. what does `null` status mean?)

Reply:
No, this is not returned to the client

Reply:
Just want to double check, I don't see `FileNotFoundException` being caught by the callers? ie. `processSyncPath`

Reply:
`FileNotFoundException` extends from `IOException` which is in the method definition of the caller `loadMetadataForPath` as a part of it's `throws` declarations.

`IOException` is then caught by the try-catch block on line 422 in `processSyncPath` or if the root path is missing, then it is caught in the handler on line 300